src/main/java/net/minecraftforge/registries/GameData.java

```diff
@@ -615,6 +620,38 @@ public void onCreate(IForgeRegistryInternal<Feature<?>> owner, RegistryManager s
         }
     }
 
+    private static class PointOfInterestTypeCallbacks implements IForgeRegistry.AddCallback<PointOfInterestType> , IForgeRegistry.ClearCallback<PointOfInterestType>, IForgeRegistry.CreateCallback<PointOfInterestType>
+    {
+        static final PointOfInterestTypeCallbacks INSTANCE = new PointOfInterestTypeCallbacks();
+        
+        @Override
+        public void onAdd(IForgeRegistryInternal<PointOfInterestType> owner, RegistryManager stage, int id, PointOfInterestType obj, @Nullable PointOfInterestType oldObj)
+        {
+            Map<BlockState, PointOfInterestType> map = owner.getSlaveMap(BLOCKSTATE_TO_POINT_OF_INTEREST_TYPE, Map.class);
+            if (oldObj != null)
+            {
+                oldObj.getBlockStates().forEach(map::remove);
+            }
+            obj.getBlockStates().forEach((state) -> 
+            {
+                if (map.put(state, obj) != null)
+                {
+                    throw Util.pauseDevMode(new IllegalStateException(String.format("%s is defined in too many tags", state)));
+                }
+            });
+        }
+
+        @Override public void onClear(IForgeRegistryInternal<PointOfInterestType> owner, RegistryManager stage)
```

Comment:
Formatting: Put the annotation on its own line like everywhere else.

Reply:
Silly IDE formatting...